File: Reverse_Words_III/reverse.js
/**
 * @param {string} word - The word to reverse
 * @returns {string} returns the word in reversal
 */
const reverse = (word) => {
	let [i, j] = [0, word.length - 1];
	const charArr = word.split("");

	while (i < j) {
		[charArr[i], charArr[j]] = [charArr[j], charArr[i]];
		i++;
		j--;
	}

	return charArr.join("");
};
/**
 * @param {string} s - The string to reverse within word
 * @returns {string} returns the string in original order but within words is reverse
 */
var reverseWords = function (s) {
	const words = s.split(" ");

	for (let i = 0; i < words.length; i++) {
		let reverseWord = reverse(words[i]);
		words[i] = reverseWord;
	}

	return words.join(" ");
};

File: Reverse_Words_III/reverse.py
class Solution:
    def reverseWords(self, s: str) -> str:

        words = s.split(" ")
        i = 0

        while i < len(words):
            revword = words[i][::-1]
            words[i] = revword
            i += 1

        return " ".join(words)
